### DISCUSSION TITLE
LDAP implementation questions and ideas

### DISCUSSION THREAD

**Discussion by jamesalbert** [semaphoreui/semaphore#970] [Ideas]
Hello 👋🏻 

I plugged semaphore into our AD server, and after piecing together info from various issues/PRs, I think I have the expected workflow:

- AD user logs in
- they are now in the users list
- admin can add to projects

Question: I was wondering if I can get confirmed that this is in fact the expected workflow.

Idea: While this works, it is a little inconvenient relative to other ldap integrations. It would be really helpful if there was additional configuration for a role based matrix that maps groups to projects (in addition to whether they are admin or not). That way, by the time the ldap user logs in the first time, their access is already setup for them.

Idea: I ran into this issue #419 because my service account I use for setup (and the group that it's in) both have spaces in their names. This led to [setup.go no parsing stdin correctly](https://github.com/ansible-semaphore/semaphore/blob/develop/cli/setup/setup.go#L174). I made a sketchy fix that I used to get going, but it's not very good. It looks like folks here have [run into a similar issue, but in a different way](https://github.com/ansible-semaphore/semaphore/pull/395). I was hoping we could find a similar solution (I'm happy to submit a PR for this if wanted).

**Comment by fiftin** [comment]
Hi @jamesalbert ,

PRs welcome! I'm not familiar with LDAP in depth, I will be glad to help.
